### ISSUE TITLE
no-unused-declaration false positive for namespace import

### ISSUE THREAD

**Issue by danvk** [cartant/eslint-plugin-etc#17]
```
$ npm install firebase-admin
```

```ts
import admin from 'firebase-admin';
//     ~~~~~ "admin" is not used; unused declarations are forbidden.
//           eslintetc/no-unused-declaration

export async function getToken(
  firebaseAuth: admin.auth.Auth,
) {
  return firebaseAuth.createCustomToken('user');
}
```

The import is clearly used, so it should not be flagged as an unused import.

The workaround isn't too bad, just import `auth` instead of the whole namespace:

```ts
import {auth} from 'firebase-admin';

export async function getToken(
  firebaseAuth: auth.Auth,
) {
  return firebaseAuth.createCustomToken('user');
}
```

If it's relevant, I have `"esModuleInterop": true` set in my `tsconfig.json`.
